clients/vscode-hlasmplugin/src/test/suite/testHelper.ts

```diff
@@ -193,9 +198,14 @@ export async function waitForDiagnosticsChange(file: string | vscode.Uri, action
             const forFile = e.uris.find(v => v.toString() === fileUri.toString());
             if (!forFile)
                 return;
-            const diags = vscode.languages.getDiagnostics(forFile);
+            let diags = vscode.languages.getDiagnostics(forFile);
             if (diags.length === initialDiags.length && diags.map(x => JSON.stringify(x)).sort().every((x, i) => x === initialDiags[i]))
                 return;
+            if (source) {
+                diags = diags.filter(d => { return d.source === source });
+                if (diags.length === 0)
+                    return;
+            }
```

Comment:
```suggestion
            const diags = vscode.languages.getDiagnostics(forFile);
            if (diags.length === initialDiags.length && diags.map(x => JSON.stringify(x)).sort().every((x, i) => x === initialDiags[i]))
                return;
            if (source && diags.every(d => d.source !== source))
                    return;
```

however I think that if the source is specified all the other code should probably use it as well? The initialialDiags should be filtered, the count test in the callback...